src/main/scala/io/renku/jsonld/JsonLD.scala

```diff
@@ -198,7 +209,32 @@ object JsonLD {
   private[jsonld] final case class JsonLDArray(jsons: Seq[JsonLD])
       extends JsonLD
       with JsonLDArrayFlatten
-      with JsonLDArrayMerge {
+      with EntitiesMerger {
+
+    override type T = JsonLD
+
+    override lazy val merge: Either[MalformedJsonLD, T] = {
+
+      def isFlatten(jsons: Seq[JsonLD]): Boolean = jsons.exists {
+        case _: JsonLDEdge => true
+        case _ => false
+      }
+
+      lazy val collectEntityLikeEntities: Seq[JsonLD] => (List[JsonLD], List[JsonLDEntityLike]) =
```

Comment:
It's too bad that there is no "typed span" on a `List`. I think cats has some utility for it 
```scala
_.partitionEither {
      case a: JsonEntityLike => Right(a)
      case a => Left(a)
    }
```
Ofc, it's the same thing at the end!